VsViewer/vs_calc/CPT.py

```diff
@@ -114,23 +161,42 @@ def calc_cpt_params(self):
         pa = 0.1
         # compute vertical stress profile
         totalStress = np.zeros(len(self.depth))
+        totalStress[0] = self.gamma[0] * self.depth[0]
         u0 = np.zeros(len(self.depth))
         for i in range(1, len(self.depth)):
             totalStress[i] = (
                 self.gamma[i] * (self.depth[i] - self.depth[i - 1]) + totalStress[i - 1]
             )
             if self.depth[i] >= self.ground_water_level:
-                u0[i] = 0.00981 * (self.depth[i] - self.depth[i - 1]) + u0[i - 1]
+                u0[i] = 0.00981 * (self.depth[i] - self.ground_water_level)
+
         effStress = totalStress - u0
-        effStress[0] = effStress[1]  # fix error caused by dividing 0
 
-        n = 0.381 * self.Ic + 0.05 * (effStress / pa) - 0.15
+        Fr = (self.Fs / (self.qt - totalStress)) * 100
+        Qtn = np.zeros(len(self.depth))
+        Ic = np.zeros(len(self.depth))
+        # assumed initial value of n
+        n = 0.5 * np.ones(len(self.depth))
+
+        # loop over each depth point
         for i in range(0, len(n)):
-            if n[i] > 1:
-                n[i] = 1
-        Qtn = ((self.qt - totalStress) / pa) * (pa / effStress) ** n
+            deltan = 1
+            # iterate Qtn, Ic and n until convergence
+            while deltan >= 0.01:
```

Comment:
do we need a reference for the relation used in this function?

Reply:
No idea if there is a reference for this convergence for Qc Ic and n. @lee-robin do you know?

Reply:
References that including Soil behavior type (equations for n, Ic, Qtn ...) = P.K. Robertson (1990, 2009, 2010); Robertson and Wride (1998)